modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -4042,9 +4073,61 @@ private void assignPartitionsStates() {
             throw new IgniteException("Failed to assign partition states", e);
         }
 
+        if (log.isInfoEnabled() && !F.isEmpty(supplayInfoMap))
+            printPartitionRebalancingFully(supplayInfoMap);
+
         timeBag.finishGlobalStage("Assign partitions states");
     }
 
+    /**
+     * Prints detail information about partitions which did not have reservation
+     * history enough for historical rebalance.
+     *
+     * @param supplayInfoMap Map contains information about supplying partitions.
+     */
+    private void printPartitionRebalancingFully(Map<String, List<SupplyPartitionInfo>> supplayInfoMap) {
+        if (hasPartitonToLog(supplayInfoMap, false)) {
+            log.info("Partitions weren't any history reservation: [" +
```

Comment:
weren't **in** any ?
I'd say "Partitions weren't present in any history reservation".